openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py

```diff
@@ -300,19 +337,90 @@ def _get_component_from_usage_key(self, usage_key):
 
         return component
 
-    def _lookup_asset_url(self, block: XBlock, asset_path: str) -> str | None:  # pylint: disable=unused-argument
+    def _get_component_version_from_block(self, block):
+        usage_key = block.usage_key
+        component = self._get_component_from_usage_key(usage_key)
+
+        block_version = block._runtime_requested_version  # pylint: disable=protected-access
+        if block_version == LatestVersion.DRAFT:
+            component_version = component.versioning.draft
+        elif block_version == LatestVersion.PUBLISHED:
+            component_version = component.versioning.published
+        else:
+            component_version = component.versioning.version_num(block_version)
+
+        return component_version
```

Comment:
Do you think it makes sense to combine this with the similar logic in `get_block()`?

https://github.com/openedx/edx-platform/blob/ecbedbcf78cd74c907f18d76b77a64e31ccf78fe/openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py#L183-L193

Reply:
I'm kind of on the fence about it. When I took a brief sketch at extracting out the common part here, the helper naming and indirection just seemed to add more confusion than it was worth. I don't feel strongly about it, but I'm inclined to just leave the bit of redundancy.

Reply:
Fine with me!